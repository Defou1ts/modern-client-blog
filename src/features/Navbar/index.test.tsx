import React from 'react';

import { render, screen } from '@testing-library/react';
import { useRouter } from 'next/router';

import renderer from 'react-test-renderer';

import { Navbar } from './index';

import type { Route } from '@shared/interfaces/route';
import type { NextRouter } from 'next/router';

const mockRoutes: Route[] = [
	{
		translationPath: 'home',
		path: '/',
	},
	{
		translationPath: 'blog',
		path: '/blog',
	},
	{
		translationPath: 'about',
		path: '/contact',
	},
	{
		translationPath: 'contact',
		path: '/about',
	},
];

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}));

describe('Navbar', () => {
	beforeAll(() => {
		const useRouterMock = {
			locale: 'en',
			locales: ['en', 'ru'],
			pathname: '/',
		};
		(useRouter as jest.Mock<NextRouter>).mockReturnValue(useRouterMock as NextRouter);
	});

	it('renders a Navbar', () => {
		render(<Navbar routes={mockRoutes} />);

		const contactText = screen.getByText('header.contact');
		expect(contactText).toBeInTheDocument();
		expect(contactText).toBeVisible();
	});

	it('Navbar snapshot', () => {
		const component = renderer.create(<Navbar routes={mockRoutes} />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
