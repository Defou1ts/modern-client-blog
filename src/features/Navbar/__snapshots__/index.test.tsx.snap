// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Navbar snapshot 1`] = `
<nav
  className="nav"
>
  <a
    className="navlink active"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    header.home
  </a>
  <a
    className="navlink"
    href="/blog"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    header.blog
  </a>
  <a
    className="navlink"
    href="/contact"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    header.about
  </a>
  <a
    className="navlink"
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    header.contact
  </a>
</nav>
`;
